cpp/src/bitmask/null_mask.cu

```diff
@@ -372,6 +458,113 @@ cudf::size_type count_unset_bits(bitmask_type const *bitmask, size_type start,
   return (num_bits - detail::count_set_bits(bitmask, start, stop, stream));
 }
 
+std::vector<size_type> count_set_bits(bitmask_type const* bitmask,
+                                      std::vector<size_type> const& indices,
+                                      cudaStream_t stream) {
+  CUDF_EXPECTS(indices.size() % 2 == 0, "Invalid ranges.");
+
+  if ((bitmask == nullptr) || (indices.size() == 0)) {
+    return std::vector<size_type>();
+  }
+
+  size_type num_ranges = indices.size() / 2;
+  thrust::host_vector<size_type> h_first_indices(num_ranges);
+  thrust::host_vector<size_type> h_last_indices(num_ranges);
+  for (size_type i = 0; i < num_ranges; i++) {
+    auto begin = indices[i * 2];
+    auto end = indices[i * 2 + 1];
+    CUDF_EXPECTS(begin >= 0, "Invalid range.");
+    CUDF_EXPECTS(end >= begin, "Invalid range.");
+    h_first_indices[i] = begin;
+    h_last_indices[i] = end;
```

Comment:
Going off of the idea of converting the end indices to be inclusive, this transformation could be done here.